amethyst_renderer/src/formats/texture.rs

```diff
@@ -7,71 +7,104 @@ use amethyst_assets::{
     ProgressCounter, Result, ResultExt, SimpleFormat,
 };
 use amethyst_core::specs::prelude::{Entity, Read, ReadExpect};
-use gfx::format::{ChannelType, SurfaceType};
+use gfx::format::{ChannelType, SurfaceType, SurfaceTyped};
 use gfx::texture::SamplerInfo;
 use gfx::traits::Pod;
 use tex::{Texture, TextureBuilder};
+use types::SurfaceFormat;
 use Renderer;
 
-/// Texture metadata, used while loading
+/// Additional texture metadata that can be passed to the asset loader or added to the prefab.
+///
+///
 #[derive(Debug, Clone, Deserialize, Serialize)]
-#[serde(default)]
 pub struct TextureMetadata {
-    /// Sampler info
-    pub sampler: Option<SamplerInfo>,
-    /// Mipmapping
-    pub mip_levels: Option<u8>,
-    /// Texture size
-    pub size: Option<(u16, u16)>,
+    /// The sampler info describes how to read from the texture, thus specifies
+    /// filter and wrap mode.
+    /// The default is nearest filtering (`FilterMethod::Scale`) and clamping (`WrapMode::Clamp`).
+    #[serde(with = "serde_helper")]
+    #[serde(default = "serde_helper::default_sampler")]
+    pub sampler: SamplerInfo,
+    /// Mipmapping levels. The default is one level.
+    #[serde(default = "serde_helper::default_mip_levels")]
+    pub mip_levels: u8,
     /// Dynamic texture
+    #[serde(default)]
     pub dynamic: bool,
-    /// Surface type
-    pub format: Option<SurfaceType>,
-    /// Channel type
-    pub channel: Option<ChannelType>,
+    /// The surface type of the texture which describes the number of color channels and their size.
+    /// In simpler words, this defines the color format, e.g. RGBA 32-bit.
+    ///
+    /// The default is `R8_G8_B8_A8`.
+    #[serde(default = "SurfaceFormat::get_surface_type")]
+    pub format: SurfaceType,
+    /// The channel type which describes the data format of the channels (e.g. how the red value
+    /// is stored).
+    ///
+    /// This is usually `Srgb` for color textures, normalmaps & similar mostly use `Unorm`
+    /// (a floating-point number between `0.0` and `1.0`.
+    pub channel: ChannelType,
```

Comment:
Note: this is the only field here that has no default.

Reply:
Maybe `Unorm` could be default. If it is possible to calculate later then `Option<ChannelType>` could be used. with `None` meaning is "get from asset or derive from usage or use `Unorm` as default"

Reply:
I un-implemented `Default` on purpose. I want to force us to immediately specify the channel type because later the information is not available anymore.